### PR TITLE
:bug: replace pkg/runtime/log for pkg/log in just the v2 scaffolding.

### PR DESCRIPTION
**Pull request by camilamacedo86**
Closes: #1269

### CODE REVIEW

File: pkg/scaffold/v2/webhook/webhook.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ package {{ .Resource.Version }}
 
 import (
 	ctrl "sigs.k8s.io/controller-runtime"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
```
Comment (camilamacedo86) [inline pkg/scaffold/v2/webhook/webhook.go]: @mengqiy done 👍

**Comment by camilamacedo86** [comment]
Oks, I will do that @mengqiy. I thought that we should not do it for v1.

**Comment by mengqiy** [comment]
> I thought that we should not do it for v1.

It should be very low risk doing that, since (1) the the deprecated package redirect to the new package. (2) Not many users use v1 any more.

The point here is mostly that we should not scaffold project using a deprecated package, which causing it doesn't pass staticcheck linter.

**Comment by camilamacedo86** [comment]
Hi @mengqiy 

> It should be very low risk doing that, since (1) the the deprecated package redirect to the new package. (2) Not many users use v1 any more.
> 
> The point here is mostly that we should not scaffold project using a deprecated package, which causing it doesn't pass staticcheck linter.

My comment was because of:

Screenshot 2020-01-17 at 19 42 47

**Also, see that after this PR be merged any V2 project will NOT be scaffolded with the deprecated lib. So, if the user is using the lint it should pass in the test/check**

Regards:

> Besides pkg/scaffold/v2/webhook/webhook.go, there are 3 more places need to be change:
> 
> pkg/scaffold/v1/controller/controller.go
> pkg/scaffold/v1/manager/cmd.go
> pkg/scaffold/v1/webhook/server.go
> Though they are for v1 project, but the idea is to not generating something using a deprecated package.

I do not think that it will be possible. The V1 is using the vendor and other versions. So, for the version used in V1 the lib is not deprecated and the new implementation is not available. 

Screenshot 2020-01-17 at 19 56 25

```
[[projects]]
  digest = "1:fbd9138a454764d4037f8edeb7bdea3f4ef4ad24b672a52cceff2a9609aacf91"
  name = "sigs.k8s.io/controller-runtime"
  packages = [
    "pkg/cache",
    "pkg/cache/internal",
    "pkg/client",
    "pkg/client/apiutil",
    "pkg/client/config",
    "pkg/controller",
    "pkg/event",
    "pkg/handler",
    "pkg/internal/controller",
    "pkg/internal/controller/metrics",
    "pkg/internal/objectutil",
    "pkg/internal/recorder",
    "pkg/leaderelection",
    "pkg/manager",
    "pkg/metrics",
    "pkg/patch",
    "pkg/predicate",
    "pkg/reconcile",
    "pkg/recorder",
    "pkg/runtime/inject",
    "pkg/runtime/log",
    "pkg/runtime/signals",
    "pkg/source",
    "pkg/source/internal",
    "pkg/webhook/admission",
    "pkg/webhook/admission/types",
    "pkg/webhook/internal/metrics",
    "pkg/webhook/types",
  ]
  pruneopts = "T"
  revision = "f1eaba5087d69cebb154c6a48193e6667f5b512c"
  version = "v0.1.12"
```

**See that in V2 we are using an upper version:**

```go
go 1.13

require (
	github.com/go-logr/logr v0.1.0
	github.com/onsi/ginkgo v1.8.0
	github.com/onsi/gomega v1.5.0
	k8s.io/apimachinery v0.0.0-20190913080033-27d36303b655
	k8s.io/client-go v0.0.0-20190918160344-1fbdaa4c8d90
	sigs.k8s.io/controller-runtime v0.4.0
)
```

This changed was made/introduced in the version [v0.2.0-alpha.0](https://github.com/kubernetes-sigs/controller-runtime/releases?after=v0.2.0-beta.2). See the PR: https://github.com/kubernetes-sigs/controller-runtime/pull/290. 

Please, let me know if it makes sense or you and if has anything else that you would like to be done for we close this one.

**Comment by mengqiy** [comment]
You are right!
Just checked what we scaffold for v1, V1 scaffolding is using controller-runtime v0.1.8, which is not a problem here. Because the log package moving and old path deprecation happened in v0.2.0+.

**Pull request merged**